cdap-ui/app/cdap/components/ProgramTable/index.js

```diff
@@ -14,47 +14,98 @@
 import shortid from 'shortid';
 import {humanReadableDate} from 'services/helpers';
-require('./ProgramTable.scss');
+import EntityIconMap from 'services/entity-icon-map';
 import T from 'i18n-react';
 import isEmpty from 'lodash/isEmpty';
+import classnames from 'classnames';
+import moment from 'moment';
+require('./ProgramTable.scss');
+
+export default class ProgramTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entities: []
+    };
+
+    this.tableHeaders = [
+      {
+        property: 'name',
+        label: T.translate('features.ViewSwitch.nameLabel')
+      },
+      {
+        property: 'programType',
+        label: T.translate('features.ViewSwitch.typeLabel')
+      },
+      // have to convert latestRun back from string to seconds from epoch
+      {
+        property: 'latestRun',
+        label: T.translate('features.ViewSwitch.ProgramTable.lastStartedLabel'),
+        sortFunc: (entity) => { return moment(entity.latestRun.start).valueOf(); }
+      },
+      {
+        property: 'status',
+        label: T.translate('features.ViewSwitch.ProgramTable.statusLabel')
+      },
+      // empty header label for Actions column
+      {
+        label: ''
+      }
+    ];
+  }
 
-export default function ProgramTable({programs}) {
-  let entities = programs.map(prog => {
-    return Object.assign({}, prog, {
-      applicationId: prog.app,
-      programType: prog.type,
-      type: 'program',
-      id: prog.id,
-      uniqueId: shortid.generate()
+  componentWillMount() {
+    let entities = this.updateEntities(this.props.programs);
+    this.setState({
+      entities
     });
-  });
-  return (
-    <div className="program-table">
-      <table className="table table-bordered">
-      <thead>
-        <tr>
-          <th>{T.translate('features.ViewSwitch.nameLabel')}</th>
-          <th>{T.translate('features.ViewSwitch.typeLabel')}</th>
-          <th>{T.translate('features.ViewSwitch.ProgramTable.lastStartedLabel')}</th>
-          <th>{T.translate('features.ViewSwitch.ProgramTable.statusLabel')}</th>
-          <th>{T.translate('features.ViewSwitch.actionsLabel')}</th>
-        </tr>
-      </thead>
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let entities = this.updateEntities(nextProps.programs);
+    this.setState({
+      entities
+    });
+  }
+
+  updateEntities(programs) {
+    let entities = programs.map(prog => {
+      return Object.assign({}, prog, {
+        applicationId: prog.app,
+        programType: prog.type,
+        type: 'program',
+        id: prog.id,
+        uniqueId: shortid.generate()
+      });
+    });
+    return entities;
+  }
+
+  renderTableBody(entities) {
+    return (
       <tbody>
         {
           entities.map(program => {
+            let icon = EntityIconMap[program.programType];
+            let statusClass = program.status === 'RUNNING' ? 'text-success' : '';
             return (
               <tr key={program.name}>
```

Comment:
@tonybach can you modify this to be `program.uniqueId`? I think that will fix this JIRA: https://issues.cask.co/browse/CDAP-8674

Do verify it though.